wrappers/binaries.go

```diff
@@ -21,13 +21,57 @@
 package wrappers
 
 import (
+	"io"
 	"os"
+	"regexp"
+	"strconv"
+	"strings"
 
 	"github.com/snapcore/snapd/dirs"
 	"github.com/snapcore/snapd/osutil"
 	"github.com/snapcore/snapd/snap"
 )
 
+func detectLegacy() (noCompletion bool, legacyCompletion bool) {
```

Comment:
could use a doc commend describing the reason to run this detection, eg.
```
// detectLegacy check whether the bash completion support is recent enough
// such that the completion files can be placed under an arbitrary path listed in
// XDG_DATA_DIRS.
```

Reply:
Done